### DISCUSSION TITLE
Batching of jobs, priorities/makespan

### DISCUSSION THREAD

**Discussion by Medwar19** [google/or-tools#3645] [CP-SAT questions] [answered]
Hi experts,

I need help understanding how to expand the job shop examples to add batching of jobs:

- How to ensure at most N customer orders are worked on in parallel?
- How to get the min/max intervals for each order and add cumulative constraint to limit the number of orders worked in in parallel?
Thanks for your suggestions!

![image](https://user-images.githubusercontent.com/18273001/213807994-42e6dfa5-2d47-4ddd-b2db-1c3a907d16f7.png)

**Comment by lperron** [comment, marked as answer]
For each order: create a start and an end var. start = min(all start vars in order), end = max(all end vars in order). Create an interval from these.
Add these order intervals to a cumulative constraint.
